pimd/pim_nht.c

```diff
@@ -623,30 +619,16 @@ int pim_parse_nexthop_update(int command, struct zclient *zclient,
 	struct interface *ifp1 = NULL;
 	struct vrf *vrf = vrf_lookup_by_id(vrf_id);
 	struct pim_instance *pim;
+	struct zapi_route nhr;
 
 	if (!vrf)
 		return 0;
 	pim = vrf->info;
 
-	s = zclient->ibuf;
-	memset(&p, 0, sizeof(struct prefix));
-	p.family = stream_getw(s);
-	p.prefixlen = stream_getc(s);
-	switch (p.family) {
-	case AF_INET:
-		p.u.prefix4.s_addr = stream_get_ipv4(s);
-		break;
-	case AF_INET6:
-		stream_get(&p.u.prefix6, s, 16);
-		break;
-	default:
-		break;
-	}
+	zapi_nexthop_update_decode(zclient->ibuf, &nhr);
```

Comment:
need error checking here since this function can fail

Reply:
done